tests/e2e/performance/csi_tests/test_pvc_multi_snapshot_performance.py

```diff
@@ -1,42 +1,445 @@
+        the end time is in the provisioner pod (csi-snapshotter container)
+
+        """
+        self.log_names = {"start": [], "end": []}
+        log.info("Looking for logs pod name")
+
+        # Getting csi log name for snapshot start creation messages
+        results = self.get_csi_pod(namespace="openshift-cluster-storage-operator")
+        for line in results:
+            if "csi-snapshot-controller" in line and "operator" not in line:
+                self.log_names["start"].append(line.split()[0])
+
+        # Getting csi log name for snapshot end creation messages
+        results = self.get_csi_pod(namespace="openshift-storage")
+        for line in results:
+            if "prov" in line and self.fs_type in line:
+                self.log_names["end"].append(line.split()[0])
+
+        log.info(
+            f"The CSI logs for the test are : {json.dumps(self.log_names, indent=4)}"
+        )
+
+    def build_fio_command(self):
+        """
+        Building the FIO command that will be run on the pod before each snapshot
+
+        """
+        # Find the path that the PVC is mounted within the POD
+        path = (
+            self.pod_obj.get("spec")
+            .get("spec")
+            .get("containers")[0]
+            .get("volumeMounts")[0]
+            .get("mountPath")
+        )
+        self.fio_cmd = (
+            "fio --name=fio-fillup --rw=write --bs=4m --direct=1 --numjobs=1"
+            " --time_based=0 --runtime=36000 --ioengine=libaio --end_fsync=1"
+            f" --filename={path}/{self.file_name} --size={self.file_size}"
+            " --output-format=json"
+        )
+        log.info(f"The FIO command is : {self.fio_cmd}")
+
+    def create_snapshotclass(self, interface):
+        """
+        Creates own volumesnapshotclass
+
+        Args:
+            interface (str): Interface type used
+
+        Returns:
+            ocs_obj (obj): Snapshotclass obj instances
+
+        """
+        if interface == constants.CEPHFILESYSTEM:
+            snapshotclass_data = templating.load_yaml(
+                constants.CSI_CEPHFS_SNAPSHOTCLASS_YAML
+            )
+            snapclass_name = "pas-test-cephfs-snapshot-class"
+        else:
+            snapshotclass_data = templating.load_yaml(
+                constants.CSI_RBD_SNAPSHOTCLASS_YAML
+            )
+            snapclass_name = "pas-test-rbd-snapshot-class"
+        snapshotclass_data["metadata"]["name"] = snapclass_name
+        ocs_obj = ocs.OCS(**snapshotclass_data)
+        log.info(f"Creating new snapshot class : {snapclass_name}")
+        try:
+            created_snapclass = ocs_obj.create(do_reload=True)
+            log.debug(created_snapclass)
+        except Exception as ex:
+            err_msg = f"Failed to create new snapshot class : {snapclass_name} [{ex}]"
+            log.error(err_msg)
+            raise Exception(err_msg)
+        return ocs_obj
+
+    def create_snapshot(self, snap_num):
```

Comment:
same here, good to move it to common library

Reply:
same as above.